src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2460,17 +2537,17 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 assert(isRMW);
 
-                if (targetReg != op1Reg)
-                {
-                    assert(targetReg != op2Reg);
-
-                    GetEmitter()->emitInsSve_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, op1Reg);
-                }
-
                 HWIntrinsicImmOpHelper helper(this, intrin.op3, node);
 
                 for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
                 {
+                    if (targetReg != op1Reg)
+                    {
+                        assert(targetReg != op2Reg);
+
+                        GetEmitter()->emitInsSve_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, op1Reg);
```

Comment:
I am wondering if we should create a separate method like `emitInsSve_movprfx()` that will let us find the references and track the usage easier.

Reply:
`emitInsSve_movprfx()` would remember that a `movprfx` was used and then every emit function would then have to have a check before emitting. That's possible. Ideally I don't want to add something we need to remember to add to every function.

I have a PR I'll push tomorrow which adds `movprfx` tracking. Can we push this discussion to that one.

Reply:
sure